daemon/oci_linux.go

```diff
@@ -900,6 +900,17 @@ func (daemon *Daemon) createSpec(c *container.Container) (*specs.Spec, error) {
 	s.Process.NoNewPrivileges = c.NoNewPrivileges
 	s.Process.OOMScoreAdj = &c.HostConfig.OomScoreAdj
 	s.Linux.MountLabel = c.MountLabel
+	if s.Annotations == nil {
+		s.Annotations = make(map[string]string)
+	}
+	for k, v := range c.Config.Labels {
+		oldValue, ok := s.Annotations[k]
+		if ok {
```

Comment:
nit: `oldValue` and `ok` are only used inside the `if`, so;

```go
if oldValue, ok := s.Annotations[k]; ok {
    logrus.Warningf("Key %q already exists in Annotations, old value=%q, new value=%q", k, oldValue, v)
    continue
}
```

The warning should also mention that we're ignoring the new value




I'm not sure what would be best though:

- under what situations would an existing label exist in the annotations?
- if an existing annotation is there, is it correct to ignore the new value (even with a warning)? Or should it produce an error; instructing the user to correct the conflicting option?

In general, I'm in favour of being _strict_, and error out. If at some point in future we think it's ok to change an error into a warning, we can do so without breaking backward compatibility, whereas the opposite is not possible.

ping @mlaventure @cpuguy83 PTAL

Reply:
I'm not sure in which scenario that would happen, but I'd vote to start with an error too,

Reply:
@thaJeztah @mlaventure  Makes sense. Will change to error.